### PR TITLE
Feat1297 filter menu quantity updated according current filter

### PR DESCRIPTION
**Pull request by Blaussh**

### CODE REVIEW

File: src/app/@ansyn/menu-items/filters/components/filter-container/filter-container.component.ts [hunk 1]
```diff
@@ -66,6 +67,20 @@ export class FilterContainerComponent implements OnInit, OnDestroy {
 			this.isGotSmallListFromProvider = (<EnumFilterMetadata>this.metadataFromState).enumsFields.size <= this.config.shortFilterListLength;
 		});
 
+	filteredOverlaysChanged$: Observable<any> = this.store.select(overlaysStateSelector)
+		.withLatestFrom(this.store.select(filtersStateSelector))
+		.filter(([{ filteredOverlays, overlays }, filterState]: [IOverlaysState, IFiltersState]) => Boolean(filterState.filters.get(this.filter)))
+		.map(([{ filteredOverlays, overlays }, filterState]: [IOverlaysState, IFiltersState]) => {
```
Comment (liorze) [inline src/app/@ansyn/menu-items/filters/components/filter-container/filter-container.component.ts]: Replace with the `do` operator.

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts [hunk 2]
```diff
@@ -50,10 +50,9 @@ export class FiltersAppEffects {
 	 */
 	@Effect()
 	updateOverlayFilters$: Observable<any> = this.actions$
-		.ofType(...facetChangesActionType, CoreActionTypes.SET_FAVORITE_OVERLAYS)
```
Comment (liorze) [inline src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts]: Revert these changes.

File: src/app/@ansyn/menu-items/filters/models/metadata/boolean-filter-metadata.ts [hunk 3]
```diff
@@ -5,39 +5,55 @@ import { FilterType } from '@ansyn/core/models/case.model';
...
 	selectOnly(key: string): void {
 		this.trueProperties.value = false;
 		this.falseProperties.value = false;
 		this[key].value = true;
 	}
 
-	accumulateData(value: boolean): void {
-		if (value) {
-			this.trueProperties.count += 1;
+	accumulateData(value: boolean, calculateFiltered?: Boolean): void {
```
Comment (liorze) [inline src/app/@ansyn/menu-items/filters/models/metadata/boolean-filter-metadata.ts]: Write a different function for the `filteredCount` property.

File: src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts [hunk 4]
```diff
@@ -29,19 +30,29 @@ export class EnumFilterMetadata implements FilterMetadata {
 		});
 	}
 
-	accumulateData(value: any): void {
+	accumulateData(value: any, calculateFiltered?: Boolean): void {
```
Comment (liorze) [inline src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts]: ditto.

File: src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts [hunk 5]
```diff
@@ -29,19 +30,29 @@ export class EnumFilterMetadata implements FilterMetadata {
...
-			this.enumsFields.get(value).count = this.enumsFields.get(value).count + 1;
+			if (calculateFiltered) {
+				this.enumsFields.get(value).filteredCount = this.enumsFields.get(value).filteredCount + 1;
+			} else {
+				this.enumsFields.get(value).count = this.enumsFields.get(value).count + 1;
+			}
 		}
 	}
 
+	resetFilterCount(): void {
```
Comment (liorze) [inline src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts]: s/resetFilterCount/resetFilteredCount

File: src/app/@ansyn/menu-items/filters/components/filter-container/filter-container.component.ts (hunk 1 above)
Comment (Blaussh) [reply src/app/@ansyn/menu-items/filters/components/filter-container/filter-container.component.ts]: done
